docs/how-to-guides/develop-a-tool/create-your-own-custom-strong-type-connection.md

```diff
@@ -0,0 +1,100 @@
+# Create and Use Your Own Custom Strong Type Connection
+Connections provide a secure method for managing credentials for external APIs and data sources in Prompt flow. This guide explains how to create and use a custom strong type connection.
+
+## What is a Custom Strong Type Connection?
+A custom strong type connection in Prompt flow allows you to define a custom connection class with strongly typed keys. This provides the following benefits:
+
+* Enhanced user experience - no need to manually enter connection keys.
+* Rich intellisense experience - defining key types enables real-time suggestions and auto-completion of available keys as you work in VS Code.
+* Central location to view available keys and data types.
+
+For other connections types, please refer to [Connections](https://microsoft.github.io/promptflow/concepts/concept-connections.html).
+
+## Prerequisites
+- Please ensure that your [Prompt flow for VS Code](https://marketplace.visualstudio.com/items?itemName=prompt-flow.prompt-flow) is updated to at least version 1.2.1.
+- Please install promptflow package and ensure that its version is 0.1.0b8 or later.
+  ```
+  pip install promptflow>=0.1.0b8
+  ```
+
+## Create a custom strong type connection
+Follow these steps to create a custom strong type connection:
+
+1. Define a Python class inheriting from `CustomStrongTypeConnection`.
+  > [!Note] Please avoid using the `CustomStrongTypeConnection` class directly.
+
+2. Use the Secret type to indicate secure keys. This enhances security by scrubbing secret keys.
+
+3. Document with docstrings explaining each key.
+
+For example:
+
+```python
+from promptflow.connections import CustomStrongTypeConnection
+from promptflow.contracts.types import Secret
+
+
+class MyCustomConnection(CustomStrongTypeConnection):
+    """My custom strong type connection.
+
+    :param api_key: The api key.
+    :type api_key: Secret
+    :param api_base: The api base.
+    :type api_base: String
+    """
+    api_key: Secret
+    api_base: str = "This is a fake api base."
+
+```
+
+See [this example](https://github.com/microsoft/promptflow/blob/main/examples/tools/tool-package-quickstart/my_tool_package/tools/tool_with_custom_strong_type_connection.py) for a complete implementation.
+
+## Use the connection in a flow
+Once you've created a custom connection, use it in your flows:
```

Comment:
Once you create a custom strong type connection, here are two ways to use it in your flows: